app/contexts/AppStateContext.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { createContext, useCallback, useContext, useEffect, useRef } from 'react'
+import { AppState, AppStateStatus } from 'react-native'
+
+export type SimplifiedAppStateStatus = 'active' | 'background'
+export type AppStateListener = (status: SimplifiedAppStateStatus) => void
+
+export interface AppStateContext {
+  addListener: (listener: AppStateListener) => number
+  removeListener: (id: number) => void
+}
+
+const appStateContext = createContext<AppStateContext>(undefined as any)
+
+export function useAppStateContext (): AppStateContext {
+  return useContext(appStateContext)
+}
+
+export function AppStateContextProvider (props: React.PropsWithChildren<any>): JSX.Element | null {
+  // DO NOT use `useState`, these are meant for appstate monitoring
+  // re-render side effect is undesireable
+  const listenerRef = useRef<{ [key: number]: AppStateListener } | null>(null)
```

Comment:
```suggestion
  const listenerRef = useRef<Record<number, AppStateListener>>({})
```

keep this initialized? reducing the need for `if (listenerRef !=)`